### PR TITLE
dirty windows patch

### PR DESCRIPTION
**Pull request by PaysanCorrezien**

### CODE REVIEW

**Comment by DreamMaoMao** [comment]
Why use the /b parameter? Is it redundant?

File: init.lua [hunk 1]
```diff
@@ -62,10 +62,15 @@ end
...
 	if ya.target_family() == "windows" then
-		shell = { "dir", path }
+		shell = { "dir", "/b", path }
 	end
 
-	local i, handle = 0, io.popen(ya.shell_join(shell))
+	-- local i, handle = 0, io.popen(ya.shell_join(shell))
+	local command = ya.shell_join(shell)
+	--BUG: shell_join add single quotes to the path, which is not supported in windows cmd
+	command = command:gsub("'", '"')
```
Comment (sxyazi) [inline init.lua]: This is not safe. I will see if there are any other ways to impl it.

**Comment by PaysanCorrezien** [comment]
`dir` alone doesn't return a list of files : [MsDocs](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/dir#parameters) :
`/b | Displays a bare list of directories and files, with no additional information. The /b parameter overrides /w. `

And for the substring, `dir` doesn't accept single quote path.
I agree that the way it's working right now is not the best, maybe it can be done with the api without using external shell commands ?
I might look into it too later if you don't have time for it, but this at least work for now.
Also with this i was able to test the `keep` which work close to what i wanted in #2 .

**Comment by sxyazi** [comment]
Hi, I pushed a new commit and it should work. Needs the latest main branch of Yazi.

**Pull request merged**